### PR TITLE
Passes the form data through on enhanced pinning

### PR DESCRIPTION
**Pull request by GlynnPhillips**
The data is put into the form but for enhanced journeys the form is
hijacked and the csrf token wasn't being passed through

### CODE REVIEW

File: components/pin-button/index.js [hunk 1]
```diff
@@ -13,17 +14,17 @@ const trackPinningAction = ({ action }) =>
 		bubbles: true
 	});
 
-const findAncestor = (el, classname) => el.classList.contains(classname)
+const findAncestor = (el, selector) => el.classList.contains(selector) || el.dataset[selector] !== undefined
```
Comment (chee) [inline components/pin-button/index.js]: this feels a little eek, maybe this should be two different functions.

actually, we use `default` from polyfill.io which includes `Element.prototype.closest` so we can use that instead!

File: components/pin-button/index.js [hunk 2]
```diff
@@ -39,11 +40,13 @@ export default () => {
 			delegate.on('click', 'button[data-prioritise-button]', event => {
 				event.preventDefault();
 
+				const form = findAncestor(event.target, 'myftPrioritise');
```
Comment (chee) [inline components/pin-button/index.js]:
```suggestion
				const form = event.target.closest('[data-myft-prioritise]');
```

File: components/pin-button/index.js [hunk 3]
```diff
@@ -39,11 +40,13 @@ export default () => {
...
+				const formData = form ? serialize(form, { hash: true }) : {};
 				const { conceptId, prioritised } = event.target.dataset;
 				const wrapper = findAncestor(event.target, 'myft-pin-button-wrapper');
```
Comment (chee) [inline components/pin-button/index.js]:
```suggestion
				const wrapper = event.target.closest('.myft-pin-button-wrapper');
```

File: components/pin-button/index.js (hunk 1 above)
Comment (GlynnPhillips) [reply components/pin-button/index.js]: Good shout

**Pull request merged**